python/pyspark/pandas/data_type_ops/base.py

```diff
@@ -194,6 +194,19 @@ def _sanitize_list_like(operand: Any) -> None:
         raise TypeError("The operation can not be applied to %s." % type(operand).__name__)
 
 
+def _is_valid_for_logical_operator(right: Any) -> bool:
+    from pyspark.pandas.base import IndexOpsMixin
+
+    return (
+        isinstance(right, IndexOpsMixin)
+        and (
+            isinstance(right.spark.data_type, BooleanType)
+            or isinstance(right.spark.data_type, IntegralType)
+        )
+        or isinstance(right, (int, bool))
+    )
```

Comment:
Could you use parentheses properly when `and` and `or` are mixed?

```py
    return (
        (
            isinstance(right, IndexOpsMixin)
            and (
                isinstance(right.spark.data_type, BooleanType)
                or isinstance(right.spark.data_type, IntegralType)
            )
        )
        or isinstance(right, (int, bool))
    )
```

Reply:
updated